scipy/linalg/flapack_other.pyf.src

```diff
@@ -386,6 +386,96 @@ subroutine <prefix2c>tgsen_lwork(ijob,wantq,wantz,select,n,a,lda,b,ldb,alpha,bet
 
 end subroutine <prefix2c>tgsen_lwork
 
+subroutine <prefix2>tgsyl(trans,ijob,m,n,a,lda,b,ldb,c,ldc,d,ldd,e,lde,f,ldf,scale,dif,work,lwork,iwork,info)
+
+    ! Solves the generalized Sylvester equation:
+    ! A * R - L * B = scale * C                
+    ! D * R - L * E = scale * F
+    ! where R and L are unknown m-by-n matrices, (A, D), (B, E) and
+    ! (C, F) are given matrix pairs of size m-by-m, n-by-n and m-by-n,
+    ! respectively, with real entries. (A, D) and (B, E) must be in
+    ! generalized (real) Schur canonical form, i.e. A, B are upper quasi
+    ! triangular and D, E are upper triangular.
+    ! The solution (R, L) overwrites (C, F). 0 <= SCALE <= 1 is an output
+    ! scaling factor chosen to avoid overflow.
+
+
+    callstatement (*f2py_func)(trans,&ijob,&m,&n,a,&lda,b,&ldb,c,&ldc,d,&ldd,e,&lde,f,&ldf,&scale,&dif,work,&lwork,iwork,&info)
+    callprotoargument char*,F_INT*,F_INT*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,<ctype2>*,<ctype2>*,F_INT*,F_INT*,F_INT*
+
+    character optional,intent(in),check((*trans=='N')||(*trans=='T')) :: trans = 'N'
+    integer optional,intent(in),check(ijob>=0 && ijob<=4):: ijob = 1
+    integer intent(hide),depend(a):: m = shape(a,1)
+    integer intent(hide),depend(b):: n = shape(b,1)
+     
+    <ftype2> intent(in),dimension(m,m):: a
+    integer intent(hide),depend(a):: lda = max(shape(a,0),1)
+     
+    <ftype2> intent(in),dimension(n,n):: b
+    integer intent(hide),depend(b):: ldb = max(shape(b,0),1)
+
+    <ftype2> intent(in,copy,out,out=r),dimension(m,n):: c
+    integer intent(hide),depend(c) :: ldc = max(shape(c,0),1)
+
+    <ftype2> intent(in),dimension(m,m):: d
+    integer intent(hide),depend(d):: ldd = max(shape(d,0),1)
+     
+    <ftype2> intent(in),dimension(n,n):: e
+    integer intent(hide),depend(e):: lde = max(shape(e,0),1)
+
+    <ftype2> intent(in,copy,out,out=l),dimension(m,n):: f
+    integer intent(hide),depend(f):: ldf = max(shape(f,0),1)
+    
+    <ftype2> intent(out):: scale
+    <ftype2> intent(out):: dif
+    <ftype2> intent(hide),depend(lwork),dimension(max(1,lwork)):: work 
+    integer optional,intent(in),depend(ijob,trans,m,n),check ( ((ijob==1||ijob==2) && (*trans=='N')) ? lwork>=max(1,2*m*n):(lwork>=n||lwork==-1)):: lwork = max(1,2*m*n)
+    integer intent(out),depend(m,n),dimension(m+n+6):: iwork
+    integer intent(out):: info
+
+end subroutine <prefix2>tgsyl
+
+subroutine <prefix2>tgsyl_lwork(trans,ijob,m,n,a,lda,b,ldb,c,ldc,d,ldd,e,lde,f,ldf,scale,dif,work,lwork,iwork,info)
```

Comment:
I've read the code a bit more and indeed we don't need this query. My sincere apologies, I thought it was doing something special but it is not

```fortran
*
      IF( info.EQ.0 ) THEN
         IF( notran ) THEN
            IF( ijob.EQ.1 .OR. ijob.EQ.2 ) THEN
               lwmin = max( 1, 2*m*n )
            ELSE
               lwmin = 1
            END IF
         ELSE
            lwmin = 1
         END IF
         work( 1 ) = lwmin
```

In fact for ijob 0 or 3 it is only using 1-long `lwork`, so we can use this and keep lwork as optional indeed.

Reply:
Thanks for looking into it. I was stuck on getting lwork =1 in the tests and thought there was something wrong with my code. I will remove the _lwork routine and make lwork optional in *tgsyl.